scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,97 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str, to_unicode
+
+class PythonRobotParser():
+    def __init__(self, content):
+        try:
+            content = to_native_str(content)
+        except UnicodeDecodeError:
+            # If we found garbage or robots.txt in an encoding other than UTF-8, disregard it.
+            # Switch to 'allow all' state.
+            content = ''
+        self.rp = RobotFileParser()
+        self.rp.parse(content.splitlines()) 
+
+    def allowed(self, url, useragent):
+        try:
+            useragent = to_native_str(useragent)
+            url = to_native_str(url)
+        except UnicodeDecodeError:
+            return False
+        return self.rp.can_fetch(useragent, url)
+
+    def sitemaps(self):
+        """RobotFileParser does not support Sitemaps directive."""
+        def _get_empty_generator():
+            return
+            yield
+        return _get_empty_generator()
+
+    def crawl_delay(self, useragent):
+        """RobotFileParser does not support Crawl-delay directive for Python version < 3.6 ."""
+        try:
+            useragent = to_native_str(useragent)
+        except UnicodeDecodeError:
```

Comment:
May be worth logging a warning.

Reply:
I think we should modify the `__init__` function to take `spider` as an argument, we can use this spider reference while logging any message. Otherwise, the user might face difficulty trying to figure out which spider a log message corresponds to.

Reply:
Maybe we should pass the ``crawler``, it gives access to the spider, the settings, and some other data that developers may want to have available in their custom parsers.